Beta 15 breaks Markdown Links

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

When updating to beta 15, some markdown links are broken that were not broken on beta 14.
It seems that removing the '.md' extension on these links will resolve the issue (see https://github.com/tauri-apps/tauri-docs/blob/746c788b5025e76bd83759530431ca4216272c52/docs/api/js/classes/http.Body.md?plain=1#L33 for reference)
These seem to be if the link is referencing the same document (as shown in the code example above)

### Steps to reproduce

Include a markdown file linking to an internal markdown file.
Include the .md extension in that link


### Expected behavior

The link should work properly 

### Actual behavior

Build is not able to start, citing a broken link

### Your environment

- Public source code: https://github.com/tauri-apps/tauri-docs/blob/746c788b5025e76bd83759530431ca4216272c52/docs/api/js/classes/http.Body.md?plain=1#L33
- Public site URL:https://tauri.studio/docs/api/js/modules/http
- Docusaurus version used: Beta.15 (sites linked are using Beta.14 due to build error)
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Node 16.13.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): macOS 12.1


### Reproducible demo

https://codesandbox.io/s/loving-easley-7be78

### Self-service

- [ ] I'd be willing to fix this bug myself.

```
7:42:29 PM: [ERROR] Unable to build website for locale en.
7:42:29 PM: [ERROR] Error: Docusaurus found broken links!
7:42:29 PM: 
7:42:29 PM: Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.
7:42:29 PM: Exhaustive list of all broken links found:
7:42:29 PM: 
7:42:29 PM: - On source page path = /docs/api/js/classes/http.Body:
7:42:29 PM:    -> linking to http.Body.md (resolved as: /docs/api/js/classes/http.Body.md)
```

Pull request that generated this log: https://github.com/tauri-apps/tauri-docs/pull/377

File/Line that the error is referencing: https://github.com/tauri-apps/tauri-docs/blob/5351725e281e66beb18a7f47876fe36e50465a0a/docs/api/js/classes/http.Body.md?plain=1#L33

--- a/packages/docusaurus-utils/src/__tests__/markdownLinks.test.ts
+++ b/packages/docusaurus-utils/src/__tests__/markdownLinks.test.ts
@@ -21,6 +21,7 @@ describe('replaceMarkdownLinks', () => {
           '@site/docs/intro.md': '/docs/intro',
           '@site/docs/foo.md': '/doc/foo',
           '@site/docs/bar/baz.md': '/doc/baz',
+          '@site/docs/http.foo.md': '/doc/http',
         },
         fileString: `
 [foo](./foo.md)
@@ -29,6 +30,7 @@ describe('replaceMarkdownLinks', () => {
 [http](http://github.com/facebook/docusaurus/README.md)
 [https](https://github.com/facebook/docusaurus/README.md)
 [asset](./foo.js)
+[looks like http...](http.foo.md)
 [nonexistent](hmmm.md)
 `,
       }),
@@ -51,6 +53,7 @@ describe('replaceMarkdownLinks', () => {
       [http](http://github.com/facebook/docusaurus/README.md)
       [https](https://github.com/facebook/docusaurus/README.md)
       [asset](./foo.js)
+      [looks like http...](/doc/http)
       [nonexistent](hmmm.md)
       ",
       }

--- a/packages/docusaurus-utils/src/markdownLinks.ts
+++ b/packages/docusaurus-utils/src/markdownLinks.ts
@@ -64,7 +64,8 @@ export function replaceMarkdownLinks<T extends ContentPaths>({
     // Replace inline-style links or reference-style links e.g:
     // This is [Document 1](doc1.md) -> we replace this doc1.md with correct link
     // [doc1]: doc1.md -> we replace this doc1.md with correct link
-    const mdRegex = /(?:(?:\]\()|(?:\]:\s?))(?!https?)([^'")\]\s>]+\.mdx?)/g;
+    const mdRegex =
+      /(?:(?:\]\()|(?:\]:\s?))(?!https?:\/\/)([^'")\]\s>]+\.mdx?)/g;
     let mdMatch = mdRegex.exec(modifiedLine);
     while (mdMatch !== null) {
       // Replace it to correct html link.

